addons/account/models/mail_message.py

```diff
@@ -78,6 +88,67 @@ def _compute_account_audit_log_preview(self):
             )
             message.account_audit_log_preview = audit_log_preview
 
+    def _search_account_audit_log_preview(self, operator, value):
+        if operator not in ['ilike'] or not isinstance(value, str):
+            raise UserError(self.env._('Operation not supported'))
+        
+        return Domain('message_type', '=', 'notification') & Domain.OR([
+        [('tracking_value_ids.old_value_char', operator, value)],
+        [('tracking_value_ids.new_value_char', operator, value)],
+        [('tracking_value_ids.old_value_text', operator, value)],
+        [('tracking_value_ids.new_value_text', operator, value)],
+    ])
+    
+    def _get_extra_ids(self):
+        # Note: Under standard record rules, self.env['account.move.line'] is already limited
+        partner_ids = self.env['account.move.line'].sudo().search([
+            ('partner_id', '!=', False),
+            ('company_id.check_account_audit_trail', '=', True),
+        ]).mapped('partner_id').ids
+
+        tax_ids = self.env['account.move.line'].sudo().search([
+            ('tax_line_id', '!=', False),
+            ('company_id.check_account_audit_trail', '=', True),
+        ]).mapped('tax_line_id').ids
+
+        company_ids = self.env['account.move.line'].sudo().search([
+            ('company_id.check_account_audit_trail', '=', True),
+        ]).mapped('company_id').ids
+
+        return {
+            'res.partner': partner_ids,
+            'account.tax': tax_ids,
+            'res.company': company_ids,
+        }
+
+    def _compute_account_audit_log_activated(self):
+        self['account_audit_log_activated'] = False
+        
+        extra_ids = self._get_extra_ids()
+        for model in DOMAINS:
+            messages_of_related = self.filtered(lambda m: m.model == model and m.res_id)
+            if messages_of_related:
+                domain = DOMAINS[model](operator='=', value=True, extra_ids=extra_ids.get(model, None))
+                related_recs = self.env[model].sudo().search([('id', 'in', messages_of_related.mapped('res_id'))] + domain)
+                recs_by_id = {record.id: record for record in related_recs}
+                for message in messages_of_related:
+                    if message.message_type == 'notification' and recs_by_id.get(message.res_id, False):
+                        message.account_audit_log_activated = True
+
+    def _search_account_audit_log_activated(self, operator, value):
+        if operator not in ('in', 'not in'):
```

Comment:
use a set instead of tuple :eyes:

Reply:
also i see that you modified the function but decided to remove them and change their places, i would advice to not change the places of a function since it will add a lot of diff for no reason